### PR TITLE
consensus/docs: draft ComponentDiagram.md

### PR DESCRIPTION
**Pull request by nfrisby**
Adds `ouroboros-consensus/docs/ComponentDiagram.md`

### CODE REVIEW

**Comment by nfrisby** [comment]
Rendered: https://github.com/input-output-hk/ouroboros-network/blob/nfrisby/consensus-component-diagram/ouroboros-consensus/docs/ComponentDiagram.md

**Comment by nfrisby** [comment]
I sketched this diagram while trying to think through a potential refinement for the Genesis design; I needed a simpler visual depiction of the status quo for myself, and I figured it also be nice visual aid for answering some questions that will come up in presentations.

I quickly also realized it's likely that it would benefit the Consensus team more generally. Hence spending an hour to open this PR.

cc @amesgen @dcoutts @abailly

File: ouroboros-consensus/docs/ComponentDiagram.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+This is a very high-level diagram  that indicates how the core data flow within the conceptual components of a single node's Consensus Layer.
```
Comment (dnadales) [inline ouroboros-consensus/docs/ComponentDiagram.md:1]:
```suggestion
This is a very high-level diagram that indicates how the core data flow within the conceptual components of a single node's Consensus Layer.
```

File: ouroboros-consensus/docs/ComponentDiagram.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        X3 --- |"O(X) = only occasionally"| Y3
+    end
+
+style Legend    fill:transparent,stroke:transparent
+style Consensus fill:transparent,stroke:transparent
+```
+
+The current implementation architecture is not an exact concretization of the conceptual components in this diagram.
+
+- The Diskpool component in this diagram is to blocks on disk as the Mempool component is to transactions in memory.
```
Comment (dnadales) [inline ouroboros-consensus/docs/ComponentDiagram.md:45]: Does the `Diskpool` only contain blocks? If so we should probably call this `BlocksPoolOnDisk` or something like that.

File: ouroboros-consensus/docs/ComponentDiagram.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+- The Mempool is one-to-one.
+- The Upstream Peers Adapter component in this diagram represents all mini protocol components for upstream peers (from which this node receives blocks) and the centralized BlockFetch decision logic.
+  For the purpose of this diagram, the Mint component behaves like a special, simple, trusted upstream peer.
+- The Downstream Peer Adapters component in this diagram represents all mini protocol components for downstream peers (to which this node sends blocks).
+  For the purpose of this diagram, the Wallet peer (or any other _local client_) behaves like a special, simple, trusted downstream peer.
+
+Remarks.
+
+- I didn't merge ChainSel and Diskpool in this diagram because I wanted the diagram to explicitly enumerate the three sources of blocks: Upstream Peers, the Mint, and from disk during node initialization.
```
Comment (dnadales) [inline ouroboros-consensus/docs/ComponentDiagram.md:59]: I don't know if we should include references to ourselves in the documentation. Maybe we should rephrase this.

File: ouroboros-consensus/docs/ComponentDiagram.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        Publisher --> |latest payload| Subscriber
+        Server -.-> |requested payload| Requester
+        X1[" "]
+        Y1[" "]
+        X1 --- |"T = transaction"| Y1
+        X1 --- |"B = block"| Y1
+        X1 --- |"L = ledger state"| Y1
+        X2[" "]
+        Y2[" "]
+        X2 --- |"H(X) = header part of X"| Y2
```
Comment (dnadales) [inline ouroboros-consensus/docs/ComponentDiagram.md:31]: What is the header part of a ledger state?

File: ouroboros-consensus/docs/ComponentDiagram.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        X2 --- |"U(X) = an update of X"| Y2
```
[the remark this answers is not in the document]
Comment (nfrisby) [reply ouroboros-consensus/docs/ComponentDiagram.md:32]: Yeah. See my answer to your previous comment and the `ChainUpdate` data type for what I had intended.

File: ouroboros-consensus/docs/ComponentDiagram.md (hunk 1 above)
Comment (nfrisby) [reply ouroboros-consensus/docs/ComponentDiagram.md:1]: Done.

**Comment by nfrisby** [comment]
Closing. PR input-output-hk/ouroboros-consensus#59 supplants this one.

**Pull request closed without merging**